inginious/frontend/user_manager.py

```diff
@@ -286,21 +288,37 @@ def get_auth_methods(self):
         """
         return self._auth_methods
 
-    def auth_user(self, username, password):
+    def auth_user(self, username, password, do_connect=True):
         """
         Authenticate the user in database
         :param username: Username/Login
         :param password: User password
-        :return: Returns a dict representing the user
+        :param do_connect: indicates if the user must be connected after authentification, True by default
+        :return: Returns a dict representing the user, or None if the authentication was not successful
         """
-        password_hash = self.hash_password(password)
-
         user = self._database.users.find_one(
-            {"username": username, "password": password_hash, "activate": {"$exists": False}})
+            {"username": username, "activate": {"$exists": False}})
+
+        def connect_user(username, user, do_connect):
+            return (user and self.connect_user(username, user["realname"], user["email"], user["language"],
+                                               user.get("tos_accepted", False)) if do_connect else user)
+
+        if user is None:
+            return None
+
+        if user["password"][:9] == "argon2id-":
```

Comment:
You should extract the method so that this string should only be located in one dict and the code remains agnostic.

Adding a new hashing method should only consists in adding the method and appropriate `hash`and `verify` functions, and changing the current method.